test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/TimestampTests.cs

```diff
@@ -81,15 +81,22 @@ public async Task Timestamp_Verify_WithOfflineRevocation_ReturnsCorrectFlagsAndL
                     result.HasFlag(SignatureVerificationStatusFlags.UnknownRevocation).Should().BeTrue();
 
                     var errors = logs.Where(l => l.Level == LogLevel.Error);
-                    errors.Count().Should().Be(RuntimeEnvironmentHelper.IsWindows ? 2 : 1);
 
                     if (RuntimeEnvironmentHelper.IsWindows)
                     {
+                        errors.Count().Should().Be(2);
                         errors.Should().Contain(w => w.Code == NuGetLogCode.NU3028 && w.Message.Contains("The revocation function was unable to check revocation because the revocation server could not be reached."));
                         errors.Should().Contain(w => w.Code == NuGetLogCode.NU3028 && w.Message.Contains("The revocation function was unable to check revocation for the certificate."));
                     }
+                    else if (RuntimeEnvironmentHelper.IsMacOSX)
+                    {
+                        errors.Count().Should().Be(1);
+                        errors.Should().Contain(w => w.Code == NuGetLogCode.NU3028 && w.Message.Contains("An incomplete certificate revocation check occurred."));
+                    }
                     else
                     {
+                        errors.Count().Should().Be(2);
+                        errors.Should().Contain(w => w.Code == NuGetLogCode.NU3028 && w.Message.Contains("The revocation function was unable to check revocation because the revocation server could not be reached."));
                         errors.Should().Contain(w => w.Code == NuGetLogCode.NU3028 && w.Message.Contains("unable to get certificate CRL"));
                     }
```

Comment:
Can you reuse `Test.Utility.Signing.SigningTestUtility.Assert*(…)` methods?

Reply:
Hm. I tried doing this, but the Utility uses `NU3018`, not `NU3028` (as in the results of the tests). Is this pointing to a bug, or is this simply a corner case in this particular call? Do I need to look into this?

Reply:
Create an overload which takes a NuGetCode and refactor common code into a private method?

The goal is to minimize code duplication and maximize code reuse.

Reply:
Yup! I was just making sure 3028 wasn't a mistake (looking into it, it looks like we have separate codes for timestamps and signatures, so that makes sense).

Reply:
Both [`NU3018`](https://github.com/NuGet/NuGet.Client/blob/7968d2c9ed05195d0581a5a17564602a16b9a4db/src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs#L348-L351) and [`NU3028`](https://github.com/NuGet/NuGet.Client/blob/7968d2c9ed05195d0581a5a17564602a16b9a4db/src/NuGet.Core/NuGet.Common/Errors/NuGetLogCode.cs#L398-L401) pertain to chain building failures.  `NU3018` applies to a primary signature while `NU3028` applies to a timestamp signature.  On the surface, I don't see anything wrong here with this test expecting `NU3028`.

Reply:
thanks, Damon! :)